cli/cli/helpers/grafloki/kubernetes_grafloki.go

```diff
@@ -0,0 +1,477 @@
+
+	configMapData := map[string]string{
+		"loki-datasource.yaml": fmt.Sprintf(`apiVersion: 1
+datasources:
+  - name: %v
+    type: loki
+    access: proxy
+    url: %v
+    isDefault: true
+    editable: true`, lokiServiceName, lokiHost),
+	}
+	grafanaConfigMap, err := k8sManager.CreateConfigMap(ctx,
+		graflokiNamespace,
+		grafanaDatasourceConfigMapName,
+		map[string]string{}, // empty labels
+		map[string]string{}, // empty annotations
+		configMapData)
+	shouldRemoveGrafanaConfigMap := true
+	defer func() {
+		if shouldRemoveGrafanaConfigMap {
+			if err := k8sManager.RemoveConfigMap(ctx, graflokiNamespace, grafanaConfigMap); err != nil {
+				logrus.Warnf("Attempted to remove Grafana datasource config map after an error occurred creating it but an error occurred removing it.")
+				logrus.Warnf("Manually remove Grafana datasource config map with name: %v", grafanaConfigMap.Name)
+			}
+		}
+	}()
+	if err != nil {
+		return "", stacktrace.Propagate(err, "An error occurred creating Grafana datasource configmap.")
+	}
+
+	grafanaDeployment, err := k8sManager.CreateDeployment(
+		ctx,
+		graflokiNamespace,
+		grafanaDeploymentName,
+		grafanaLabels,
+		map[string]string{}, // empty annotations
+		[]apiv1.Container{}, // no init containers
+		[]apiv1.Container{
+			{
+				Name:  "grafana",
+				Image: grafanaImage,
+				Ports: []apiv1.ContainerPort{
+					{
+						Name:          "",
+						ContainerPort: grafanaPort,
+						HostPort:      0,
+						Protocol:      "",
+						HostIP:        "",
+					},
+				},
+				Env: []apiv1.EnvVar{
+					{
+						Name:      "GF_AUTH_ANONYMOUS_ENABLED",
+						Value:     "true",
+						ValueFrom: nil,
+					},
+					{
+						Name:      "GF_AUTH_ANONYMOUS_ORG_ROLE",
+						Value:     "Admin",
+						ValueFrom: nil,
+					},
+					{
+						Name:      "GF_SECURITY_ALLOW_EMBEDDING",
+						Value:     "true",
+						ValueFrom: nil,
+					},
+				},
+				VolumeMounts: []apiv1.VolumeMount{
+					{
+						Name:             "datasources",
+						MountPath:        "/etc/grafana/provisioning/datasources",
+						ReadOnly:         false,
+						SubPath:          "",
+						MountPropagation: nil,
+						SubPathExpr:      "",
+					},
+				},
+				Command:    nil,
+				Args:       nil,
+				WorkingDir: "",
+				EnvFrom:    nil,
+				Resources: apiv1.ResourceRequirements{
+					Limits:   nil,
+					Requests: nil,
+					Claims:   nil,
+				},
+				ResizePolicy:             nil,
+				VolumeDevices:            nil,
+				LivenessProbe:            nil,
+				ReadinessProbe:           nil,
+				StartupProbe:             nil,
+				Lifecycle:                nil,
+				TerminationMessagePath:   "",
+				TerminationMessagePolicy: "",
+				ImagePullPolicy:          "",
+				SecurityContext:          nil,
+				Stdin:                    false,
+				StdinOnce:                false,
+				TTY:                      false,
+			},
+		},
+		[]apiv1.Volume{{
+			Name: "datasources",
```

Comment:
isn't hardcoding the string `datasources` here and above pretty fragile vs having named variable to link them? DRY and all that